apps/voting/app/src/vote-hooks.js

```diff
@@ -0,0 +1,90 @@
+import { useState, useEffect } from 'react'
+import { first } from 'rxjs/operators'
+
+// transforms an observable into a promise
+function promisify(observable) {
```

Comment:
This is unnecessary for the calls that use it. Given how both `external` and normal `call`s use `sendAndObserveResponses` which returns an observable, you can just `toPromise` directly on the observables without the `pipe` and `first`

https://github.com/aragon/aragon.js/blob/f864f7e1c77de27c3b46cc16d38e8dc94d7da665/packages/aragon-api/src/index.js#L173

https://github.com/aragon/aragon.js/blob/e7be88b49bae33e0320efc9abb52abde8b04d906/packages/aragon-rpc-messenger/src/index.js#L90

Reply:
That’s what I was hesitating about, thanks! Just to confirm: does `toPromise()` complete the subscriber the same way `first()` does?

Reply:
@bpierre See the edit in the original comment.

As for the question, I believe `toPromise` will return a promise that only resolves if the the observable `completes` (ends). If it doesn't end it will never complete.

Reply:
Confirmed: https://jsbin.com/munemiciyo/1/edit?js,console

Reply:
Thanks!